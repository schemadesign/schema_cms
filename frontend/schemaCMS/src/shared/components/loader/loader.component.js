import React, { PureComponent } from 'react';
import { FormattedMessage } from 'react-intl';

import messages from './loader.messages';

import { LoaderContainer } from './loader.styles';

export class Loader extends PureComponent {
  static propTypes = {};

  render() {
    return (
      <LoaderContainer>
        <FormattedMessage {...messages.loading} />
      </LoaderContainer>
    );
  }
}
