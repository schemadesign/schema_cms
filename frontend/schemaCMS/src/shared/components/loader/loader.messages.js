/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  loading: {
    id: 'shared.components.loader.loading',
    defaultMessage: 'Loading...',
  },
});
