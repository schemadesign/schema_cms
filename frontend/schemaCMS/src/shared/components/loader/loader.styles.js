import styled, { keyframes } from 'styled-components';

const SIZE = 20;

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;

export const Container = styled.div`
  display: inline-block;
  position: relative;
  width: ${SIZE}px;
  height: ${SIZE}px;

  div {
    box-sizing: border-box;
    display: block;
    position: absolute;
    width: ${SIZE}px;
    height: ${SIZE}px;
    border: ${SIZE / 8}px solid #fff;
    border-radius: 50%;
    animation: ${rotate} 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
    border-color: #fff transparent transparent transparent;
  }
  div:nth-child(1) {
    animation-delay: -0.45s;
  }
  div:nth-child(2) {
    animation-delay: -0.3s;
  }
  div:nth-child(3) {
    animation-delay: -0.15s;
  }
`;
