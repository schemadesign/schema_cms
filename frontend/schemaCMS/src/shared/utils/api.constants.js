const API_PREFIX = process.env.REACT_APP_BASE_API_URL;

export const AUTH_PATH = `${API_PREFIX}/auth/login/auth0`;
export const TOKEN_PATH = '/auth/token/';
export const USER_PATH = '/user/';
