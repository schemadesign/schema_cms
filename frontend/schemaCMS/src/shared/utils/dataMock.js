export const tableData = [
  {
    'First Name': 'George',
    'Last Name': 'Washington',
    'Full Name': 'George Washington',
    Birthplace: 'Westmoreland County',
    'State of Birth': 'VA',
    Birthday: '02/22/1732',
    Url: 'some.url.com',
  },
  {
    'First Name': 'John',
    'Last Name': 'Adams',
    'Full Name': 'John Adams',
    Birthplace: 'Braintree',
    'State of Birth': 'MA',
    Birthday: '10/30/1735',
    Url: 'some.url.com',
  },
  {
    'First Name': 'Thomas',
    'Last Name': 'Jefferson',
    'Full Name': 'Thomas Jefferson',
    Birthplace: 'Shadwell',
    'State of Birth': 'VA',
    Birthday: '04/13/1743',
    Url: 'some.url.com',
  },
  {
    'First Name': 'James',
    'Last Name': 'Madison',
    'Full Name': 'James Madison',
    Birthplace: 'Port Conway',
    'State of Birth': 'VA',
    Birthday: '03/16/1751',
    Url: 'some.url.com',
  },
  {
    'First Name': 'James',
    'Last Name': 'Monroe',
    'Full Name': 'James Monroe',
    Birthplace: 'Monroe Hall',
    'State of Birth': 'VA',
    Birthday: '04/28/1758',
    Url: 'some.url.com',
  },
];

export const tableFields = {
  'First Name': {},
  'Last Name': {},
  'Full Name': {},
  Birthplace: {},
  'State of Birth': {},
  Birthday: {},
  Url: {},
};
