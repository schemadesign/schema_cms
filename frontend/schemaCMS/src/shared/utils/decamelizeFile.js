import { decamelize } from 'humps';

export const decamelizeThatDontBreaksFile = object => {
  if (object && !(object instanceof File)) {
    if (object instanceof Array) {
      return object.map(item => decamelizeThatDontBreaksFile(item));
    }
    if (object instanceof FormData) {
      let formData = new FormData();
      for (const [key, value] of object.entries()) {
        formData.append(decamelize(key), value);
      }
      return formData;
    }
    if (typeof object === 'object') {
      return Object.keys(object).reduce(
        (acc, next) => ({
          ...acc,
          [decamelize(next)]: decamelizeThatDontBreaksFile(object[next]),
        }),
        {}
      );
    }
  }
  return object;
};
