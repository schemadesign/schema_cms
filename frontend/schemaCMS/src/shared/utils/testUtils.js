import { Router } from 'react-router';
import { IntlProvider } from 'react-intl';
import { act, create } from 'react-test-renderer';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { Theme } from 'schemaUI';
import PropTypes from 'prop-types';

import { DEFAULT_LOCALE, translationMessages } from '../../i18n';
import browserHistory from './history';

const ProvidersWrapper = ({ children }) => (
  <Router history={browserHistory}>
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <ThemeProvider theme={Theme.dark}>{children}</ThemeProvider>
    </IntlProvider>
  </Router>
);

ProvidersWrapper.propTypes = {
  children: PropTypes.node.isRequired,
};

export const makeContextRenderer = async component => {
  let wrapper = null;

  await act(async () => {
    wrapper = await create(<ProvidersWrapper>{component}</ProvidersWrapper>);
  });

  return wrapper;
};
