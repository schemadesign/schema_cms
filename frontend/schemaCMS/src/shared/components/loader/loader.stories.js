import React from 'react';
import { storiesOf } from '@storybook/react';

import { Loader } from './loader.component';

storiesOf('Shared/Loader', module).add('Default', () => <Loader />);
