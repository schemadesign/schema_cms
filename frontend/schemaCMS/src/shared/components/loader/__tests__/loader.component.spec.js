import React from 'react';

import { Loader } from '../loader.component';
import { defaultProps } from '../loader.stories';
import { makeContextRenderer } from '../../../utils/testUtils';

describe('Loader: Component', () => {
  const render = props => makeContextRenderer(<Loader {...defaultProps} {...props} />);

  it('should render correctly', async () => {
    const wrapper = await render();
    expect(wrapper).toMatchSnapshot();
  });
});
