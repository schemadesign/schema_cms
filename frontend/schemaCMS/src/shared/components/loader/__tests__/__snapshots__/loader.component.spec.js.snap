// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader: Component should render correctly 1`] = `
<div
  className="sc-bdnxRM hvTwJZ"
>
  <div />
  <div />
  <div />
  <div />
</div>
`;
